### PR TITLE
vm: doc and output related bug fixes

### LINKED ISSUE #1555: Documentation for vm extension set lacks important details
az vm extension set supports:
`    --protected-settings  : Protected settings or a file path with such contents.
    --settings            : Public settings or a file path with such contents.`

This is a really unhelpful comment.  We need to denote what the format for settings is and (ideally), and users need a way to determine what settings are available for a given extension.  (This probably can't be via the docs, but I see no way to get that information via commands.)

### LINKED ISSUE #1676: [VM] Add aliases to the parameter names of the image commands
The `az vm image` commands are great, but the parameter names can feel a little verbose after using them regularly. For example, instead of needing to do something like this:

```shell
az vm image list --all -l westus --publisher meanio --offer meanio-050-vm --sku meanio-050-vm
```

It would be awesome to be able to do:

```shell
az vm image list --all -l westus -p meanio -o meanio-050-vm -s meanio-050-vm
```

### LINKED ISSUE #1680: [VM] list-ip-addresses command doesn't support table output by default
Running `az vm list-ip-addresses -otable` results in just a blank line being printed. Explicitly setting another output type (e.g. `--osjonc`) correctly displays the IP addresses, however, it would be ideal if this command worked correctly with the table output, without needing to specify a custom `--query`.

screen shot 2017-01-07 at 7 24 23 pm

### LINKED ISSUE #1738: Remove hack in VM
Source: https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/commands.py#L106

```
#Remove the hack after https://github.com/Azure/azure-rest-api-specs/issues/352 fixed
from azure.mgmt.compute.models import ContainerService#pylint: disable=wrong-import-position
```

Issue https://github.com/Azure/azure-rest-api-specs/issues/352 has been fixed.

### PR DESCRIPTION
**Pull request by yugangw-msft**
fix #1738, fix #1680, fix #1676, fix #1555

### CODE REVIEW

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -132,6 +132,9 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
     register_cli_argument(scope, 'no_auto_upgrade', action='store_true', help='by doing this, extension system will not pick the highest minor version for the specified version number, and will not auto update to the latest build/revision number on any scale set updates in future.')
 
 register_cli_argument('vm image list', 'image_location', location_type)
+register_cli_argument('vm image', 'publisher_name', options_list=('--publisher', '-p'))
+register_cli_argument('vm image', 'offer', options_list=('--offer', '-f'))
+register_cli_argument('vm image', 'sku', options_list=('--sku', '-s'))
```
Comment (lostintangent) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: This is probably a stupid question, but does this update impact the `vm image list-offers` and `vm image list-skus` commands as well? If so, then yay!

Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: That is correct

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/commands.py [hunk 2]
```diff
@@ -20,6 +20,23 @@
...
 
+
+def transform_ip_addresses(result):
+    transformed = []
+    for r in result:
+        network = r['virtualMachine']['network']
+        public = network.get('publicIpAddresses')
+        public_ip_addresses = ','.join([p['ipAddress'] for p in public if p['ipAddress']]) if public else None
+        private = network.get('privateIpAddresses')
+        private_ip_addresses = ','.join(private) if private else None
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/commands.py]: How come you don't iterate over the list like you do above with public ip?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/commands.py [hunk 3]
```diff
@@ -20,6 +20,23 @@
...
+        entry = OrderedDict([('virtualMachine', r['virtualMachine']['name']),
+                             ('PublicIPAddresses', public_ip_addresses),
+                             ('PrivateIPAddresses', private_ip_addresses)])
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/commands.py]: lowercase names here?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/commands.py (hunk 2 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/commands.py]: private_ip_addresses has a difference schema which is just an array of  address

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/commands.py (hunk 3 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/commands.py]: fixed

**Pull request merged**